### PR TITLE
[WIP] allow to specify to drop ENTRYPOINT instruction from trait

### PR DESCRIPTION
**Pull request by mikey179**
When you use a Dockerfile which is not created as a trait but standalone like https://github.com/RobLoach/docker-composer/blob/master/base/Dockerfile you might need to be able to remove the ENTRYPOINT instruction, as the image you intend to build can't have a single ENTRYPOINT. As it seems to be impossible to remove the ENTRYPOINT inside the Dockerfile in which the other one is used it should be possible to remove it altogether.

This implementation is a proof of concept and surely needs to be discussed. Also, the DROP instruction could be extended to allow to drop other instructions from the used Dockerfile as well.

```
FROM php:7-cli

USE github.com/RobLoach/docker-composer/base:master | DROP entrypoint 

CMD ["/bin/bash"]
```

Also to be discussed: whether it's more likely you want to drop any ENTRYPOINT instruction or to keep it, e.g. what the default value should be. If the default value should be to drop the instruction probably should read KEEP ENTRYPOINT then.

### CODE REVIEW

**Comment by thekid** [comment]
By adding this, we need to be cautious not to introduce all the complexity of solutions like https://github.com/coryb/dfpp

**Comment by thekid** [comment]
You can also erase entry points as follows:

```dockerfile
FROM php:7-cli

ENTRYPOINT ["/usr/bin/php", "-a"]

ENTRYPOINT -

CMD ["/bin/bash"]
```

**Comment by mikey179** [comment]
Close, as it would introduce too much complexity which is currently not worth the effort nor the gains of what could be done with it.
Also, I found a better way for devcli by explicitly overwriting the EXTENSION statement, see https://github.com/mikey179/devcli/commit/3d5b7d92ff4e3046f3c5371d7dc6e2634caf2111.

File: use/extension.go [hunk 1]
```diff
@@ -96,5 +98,16 @@ func (s *Statement) Origin() (origin *Origin, err error) {
...
 // Extension func for parser
 func (c *Context) Extension(file *dockerfile.Dockerfile, line int, tokens *dockerfile.Tokens) dockerfile.Statement {
-	return &Statement{Context: c, Line: line, Reference: tokens.NextLine()}
+	instruction := tokens.NextLine()
+	var reference string
+	dropEntrypoint := false
+	pos := strings.LastIndex(instruction, "|")
+	if -1 == pos {
+		reference = strings.TrimSpace(instruction)
+	} else {
```
Comment (thekid) [inline use/extension.go]: Thanks for the hint with `Trim` here - applied to https://github.com/tueftler/doget/releases/tag/v0.6.0

**Pull request closed without merging**